File: Backend/src/Models/Product.model.js
const mongoose = require("mongoose")

const productSchema = new mongoose.Schema({
    productName: {
        type: String,
        required: true
    },
    productPrice: {
        type: Number,
        required: true
    },
    productShortDesc: {
        type: String,
        required: true
    },
    productRichDesc: {
        type: String
    },
    productImage: [{
        type: String
    }],
    productPublisherId: {
        type: mongoose.Schema.Types.ObjectId,
        ref: "User"
    },
    discountOnProduct: {
        type: Number,
        required: true,
        default: 0,
        min: 0,
        max: 100
    },
    taxOnProduct: {
        type: Number,
        required: true,
    }
    // category: {
    //     type: mongoose.Schema.Types.ObjectId,
    //     ref: "Category"
    // },
    // size: {
    //     type: String,
    //     default: null
    // },
    // review: [{
    //     type: mongoose.Schema.Types.ObjectId,
    //     ref: "Review"
    // }]

}, { timestamps: true })

module.exports = mongoose.model("Product", productSchema)

File: ecommerce_flutter/lib/Models/OrderScreenModel/ProductModel.dart
class ProductModel {
  String? productId;
  String? productName;
  num? productPrice;
  String? productShortDesc;
  String? productRichDesc;
  String? sellerId;
  List<dynamic>? productImage;

  ProductModel({
    required this.productId,
    required this.productName,
    required this.productPrice,
    required this.productShortDesc,
    required this.productRichDesc,
    required this.sellerId,
    required this.productImage,
  });

  ProductModel.fromJson(Map<String, dynamic> json) {
    productId = json["_id"];
    productName = json["productName"];
    productPrice = json["productPrice"];
    productShortDesc = json["productShortDesc"];
    productRichDesc = json["productRichDesc"];
    sellerId = json["productPublisherId"];
    productImage = json["productImage"];
  }

  Map<String, dynamic> toJson() {
    Map<String, dynamic> data = new Map<String, dynamic>();
    data["productId"] = productId;
    data["productName"] = productName;
    data["productPrice"] = productPrice;
    data["productRichDesc"] = productRichDesc;
    data["productShortDesc"] = productShortDesc;
    data["productPublisherId"] = sellerId;
    data["productImage"] = productImage;
    return data;
  }
}
